.github/workflows/auto-recode.yml

```diff
@@ -0,0 +1,60 @@
+name: pr-number-assign
+on: 
+  pull_request:
+permissions:
+  contents: write
+  
+jobs:
+  pr-number-assign:
+    runs-on: ubuntu-latest 
+    
+    env:
+      FEAT: default
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }}
+    
+    - name: do work
+      run: |
+        script=$(cat <<EOF
+        import os
+
+        pr_number = "${{github.event.pull_request.number}}"
+        path = "/home/runner/work/backend.ai/backend.ai/changes"
+        t_types = ["breaking","feature","enhance","deprecation","fix","doc","deps","misc"]
+        target_file = ""
+
+        files = os.listdir(path)
+        print(str(files) + " are in changes floder.")
+
+        for file in files :
+            if str(file[0:file.find(".")]) == pr_number :
+                print("Change log file already exist for this PR.")
+                exit(0)
+
+        for t_type in t_types :
+            if t_type in files : 
+                target_file = t_type
+                env_file = os.getenv('GITHUB_ENV')
+                myfile = open(env_file, "a")
+                myfile.write("FEAT=" + t_type)  
+                myfile.close()
+                os.rename(path + "/" + target_file , path + "/" + pr_number + "." + target_file + ".md")
+                print(target_file +" file changed to "+ pr_number+"."+target_file+".md")
+                exit(0)
+
+        if target_file == "":
+            print("There is not change log file for this PR in changes folder.")
+            print("Also, there is not feature named file in changes folder.")
+            exit(0)
+        
+        EOF
+        )
+        echo "$script" | python
```

Comment:
I think you could instead set the `shell` directive `"/usr/bin/env python"` to make it as a Python script.

Also, please keep the PEP-8/Black compliant codestyle even for this embedded Python script. If it becomes too long, I'd recommend to put it as a standalone script file under `scripts/` and call it from the workflow YAML, for easier testing and better editor supports.